consensus/misc/eip1559.go

```diff
@@ -102,13 +102,75 @@ func VerifyEip1559Header(config *params.ChainConfig, parent, header *types.Heade
 }
 
 // CalcBaseFee calculates the basefee of the header.
-func CalcBaseFee(config *params.ChainConfig, parent *types.Header, parentL1BaseFee *big.Int) *big.Int {
+func CalcBaseFee(config *params.ChainConfig, parent *types.Header, parentL1BaseFee *big.Int, currentHeaderTime uint64) *big.Int {
 	if config.Clique != nil && config.Clique.ShadowForkHeight != 0 && parent.Number.Uint64() >= config.Clique.ShadowForkHeight {
 		return big.NewInt(10000000) // 0.01 Gwei
 	}
 
-	scalar, overhead := ReadL2BaseFeeCoefficients()
-	return calcBaseFee(scalar, overhead, parentL1BaseFee)
+	if parent == nil || parent.Number == nil || !config.IsFeynman(currentHeaderTime) {
```

Comment:
> `parent.Number == nil`

Is this possible?

Reply:
I think it's due to some test cases, for example: https://github.com/scroll-tech/go-ethereum/blob/e2f18eebe0fdd107a79913dcaf7eade74c04d709/consensus/misc/eip1559_test.go#L127. But `parent.Number == nil` check seems to be an overkill (but also defensive).